tests/unit/lib/charms/vault_k8s/v0/test_vault_autounseal.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import textwrap
+import unittest
+from unittest.mock import patch
+
+from charms.vault_k8s.v0.vault_autounseal import (
+    VaultAutounsealDestroy,
+    VaultAutounsealDetailsReadyEvent,
+    VaultAutounsealInitialize,
+    VaultAutounsealProvides,
+    VaultAutounsealRequires,
+)
+from ops import testing
+from ops.charm import CharmBase
+
+AUTOUNSEAL_PROVIDES_RELATION_NAME = "vault-autounseal-provides"
+AUTOUNSEAL_REQUIRES_RELATION_NAME = "vault-autounseal-requires"
+
+
+class VaultAutounsealProviderCharm(CharmBase):
+    metadata_yaml = textwrap.dedent(
+        """
+        name: vault-autounseal-provider
+        containers:
+          vault:
+            resource: vault-image
+        provides:
+          vault-autounseal-provides:
+            interface: vault-autounseal
+        """
+    )
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.interface = VaultAutounsealProvides(self, AUTOUNSEAL_PROVIDES_RELATION_NAME)
+        self.framework.observe(
+            self.interface.on.vault_autounseal_initialize, self._on_vault_autounseal_initialize
+        )
+        self.framework.observe(
+            self.interface.on.vault_autounseal_destroy, self._on_vault_autounseal_initialize
+        )
+
+    def _on_vault_autounseal_initialize(self, event: VaultAutounsealInitialize):
+        pass
+
+    def _on_vault_autounseal_destroy(self, event: VaultAutounsealDestroy):
+        pass
+
+
+class VaultAutounsealRequirerCharm(CharmBase):
+    metadata_yaml = textwrap.dedent(
+        """
+        name: vault-autounseal-requirer
+        containers:
+          vault:
+            resource: vault-image
+        requires:
+          vault-autounseal-requires:
+            interface: vault-autounseal
+        """
+    )
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.interface = VaultAutounsealRequires(self, AUTOUNSEAL_REQUIRES_RELATION_NAME)
+        self.framework.observe(
+            self.interface.on.vault_autounseal_details_ready, self._on_details_ready
+        )
+
+    def _on_details_ready(self, event: VaultAutounsealDetailsReadyEvent):
+        pass
+
+
+class TestVaultAutounsealProvides(unittest.TestCase):
+    def setUp(self):
+        self.harness = testing.Harness(
+            VaultAutounsealProviderCharm, meta=VaultAutounsealProviderCharm.metadata_yaml
+        )
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def setup_relation(self, leader: bool = True) -> tuple:
+        """Set up a relation between the charm and a remote app with 1 unit."""
+        remote_app = "vault-autounseal-requires"
+        remote_unit = remote_app + "/0"
```

Comment:
```suggestion
        remote_unit =  f"{remote_app}/0"
```

Reply:
Use of f-string for consistency with other parts of the code.

Reply:
Carry over from vault_kv :laughing:  -- I'll change.

Reply:
fixed in https://github.com/canonical/vault-k8s-operator/pull/346/commits/0651467cd3a75065ba0f04391194ceef96844526